### PR TITLE
Markdown Suggestions

### PR DESCRIPTION
**Pull request by kendallgassner**
## Markdown Accessibility dev-tools 

I am working to create a markdown dev-tools validator. The validator will check any textArea that:

- is not hidden on the screen i.e. `clientWidth=0`
- has content i.e. `value != ''`

### Working Rules
 *  checks that written markdown has a heading
 *  ensures that written markdown does not use an h1 tag
 *  ensures that written markdown includes descriptive links

### Features:
- On non-github pages the validation Button and the reset button are disabled.
- The reset button will clear all of the error messages, and remove any added focus rings. 
- The Find TextArea button will highlight the markdown field that the error stems from 

A snippet of what the markdown dev-tools looks like:

Markdown accessibility dev-tools showing results of a markdown validation. The results show errors in one textareas

### Questions:
- not sure If I we should be using the Mona illustration 

### Future TODOS:

- I would love to use primer components and replace a lot of the styling in `panel.css`
- We should internationalization the non-descriptive link text array so users who do write their markdown in english can also use this feature. 
- validate image alt text
- Report bugs individually. For example we report the "Non Descriptive Links" error once ... even if multiple links are not descriptive but it would be nice to report each link individually to provide more descriptive feedback.

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  presets: [["@babel/preset-env", { targets: { node: "current" } }]],
+};
```
Comment (kendallgassner) [inline babel.config.js:3]: added for jest testing

File: devtools-feedback/devtools.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// Create a tab in the devtools area
+chrome.devtools.panels.create("GitHub Markdown a11y", "./devtools-feedback/toast.png", "./devtools-feedback/panel.html");
```
Comment (inkblotty) [inline devtools-feedback/devtools.js]: Should we use "Mona's Markdown a11y" everywhere instead of "GitHub Markdown a11y"?

**Comment by inkblotty** [comment]
🎉 👏 🥳

**Comment by khiga8** [comment]
This is so exciting! Thank you for opening this PR (and also adding tests) 👏!

Some initial thoughts/questions!

1. Is there a reason you decided to provide a validation mechanism through the dev panel versus providing immediate feedback directly on the page? (e.g. appending textual feedback onto the `Preview` editor). This is non-blocking by the way, just curious!
2. Regarding the rule: `checks that written markdown has a heading`.  I don't believe this rule applies to markdown comments but I got flagged. Is this rule appropriate? 
3. When I switch to the `Preview` tab without submitting and run the `Validate Markdown`, it doesn't seem to validate the markdown. Is that expected?
4. The links on the panel appear the same as the surrounding static text. I think we should add an underline or some other styling.

**Comment by kendallgassner** [comment]
> When I switch to the Preview tab without submitting and run the Validate Markdown, it doesn't seem to validate the markdown. Is that expected?

Right now I am querying by textArea 🤔 -- because these pages have several hidden textAreas I have had to filter by hidden textAreas... I believe the when a user selects preview the textArea becomes hidden and therefore the textArea is removed from the validation list :/ .... We might be able to optimize this by adding an data attribute of some sort to edited textAreas in the future

**Comment by kendallgassner** [comment]
> The links on the panel appear the same as the surrounding static text. I think we should add an underline or some other styling.

Thank you for catching this! I had these links styled so something might have gotten messed up let me fix  👩‍🎨

**Comment by kendallgassner** [comment]
> Is there a reason you decided to provide a validation mechanism through the dev panel versus providing immediate feedback directly on the page? (e.g. appending textual feedback onto the Preview editor). This is non-blocking by the way, just curious!

I think to avoid noise as the user is typing -- I don't want to provide feedback until the user is ready

**Comment by kendallgassner** [comment]
> regarding the rule: checks that written markdown has a heading. I don't believe this rule applies to markdown comments but I got flagged. Is this rule appropriate?

Disabling the missing heading rule for now because sometimes title fields are separate from the main textarea:

an issue form where the title of the issue has a separate input field then the overall text area

**Review by khiga8 — APPROVED** [review]
Great work!!! Let's ship it!!!! 💪🚀

**Comment by khiga8** [comment]
I would also recommend creating follow-up issues to document the future to-dos you listed!

**Pull request merged**